test/functional/y2038.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test covering the Year 2038 problem.
+
+https://en.wikipedia.org/wiki/Year_2038_problem
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+EPOCH_2038_BEFORE = 2147440306 # (GMT): Monday, January 18, 2038 3:11:46 PM
+EPOCH_2038_AFTER = 2147613106 # (GMT): Wednesday, January 20, 2038 3:11:46 PM
+
+class Y2038EpochTest(BitcoinTestFramework):
```

Comment:
A similar test for 2106 to handle an unsigned 32-bit time_t type. See https://en.bitcoin.it/wiki/Block_timestamp

Reply:
Wouldn't it be a better test that you set the time on the machine itself and let it test what value std::chrono::system_clock will generate?

Reply:
Yes. I'm going to take a stab at it.

Reply:
Depending on the approach you end up using, it might be an interesting exercise to try writing this as one of the C++ unit tests, once you have it figured out in Python.

Reply:
I'll work on the unit test. It will be interesting. 

Thank you for the review @jonatack.